src/Lie_algebra/Lie_algebra_interface.jl

```diff
@@ -0,0 +1,86 @@
+
+"""
+    LieAlgebra{𝔽, G} <: AbstractManifold{𝔽}
+
+Represent the Lie Algebra ``$(_math(:𝔤))``, that is a ``𝔽``vector space with an associated
+[`lie_bracket`](@ref) ``[⋅,⋅]: $(_math(:𝔤))×$(_math(:𝔤)) → $(_math(:𝔤))`` which fulfills
+
+1. ``[X,X] = 0`` for all ``X ∈ $(_math(:𝔤))``
+2. The Jacobi identity holds ``[X, [Y,Z]] = [[X,Y],Z] = [Y, [X,Z]]`` holds for all ``X, Y, Z ∈ $(_math(:𝔤))``.
+
+The Lie algebras considered here are those related to a [`LieGroup`](@ref) ``$(_math(:G))``,
+namely the tangent space the tangent space ``T_{$(_math(:e))}$(_math(:G))`` at the [`Identity`](@ref),
+this is internally just a `const` of the corresponding $(_link(:TangentSpace)).
+
+# Constructor
+
+    LieAlgebra(G::LieGroup)
+
+Return the Lie Algebra belonging to the [`LieGroup`](@ref) `G`.
+"""
+const LieAlgebra{𝔽,G,I} = ManifoldsBase.Fiber{
+    𝔽,ManifoldsBase.TangentSpaceType,G,I
+} where {𝔽,G<:LieGroup{𝔽},I<:Identity}
+
+function LieAlgebra(G::LieGroup{𝔽}) where {𝔽}
+    return LieAlgebra{𝔽,typeof(G),typeof(Identity(G))}(
+        G, Identity(G), ManifoldsBase.TangentSpaceType()
+    )
+end
+
+_doc_lie_bracket = """
+    lie_bracket!(𝔤::LieAlgebra, X, Y)
+    lie_bracket!(𝔤::LieAlgebra, Z, X, Y)
+
+Compute the Lie bracket ``[⋅,⋅]: $(_math(:𝔤))×$(_math(:𝔤)) → $(_math(:𝔤))`` which fulfills
+
+1. ``[X,X] = 0`` for all ``X ∈ $(_math(:𝔤))``
+2. The Jacobi identity holds ``[X, [Y,Z]] = [[X,Y],Z] = [Y, [X,Z]]`` holds for all ``X, Y, Z ∈ $(_math(:𝔤))``.
+
+The computation can be done in-place of `Z`.
+"""
+function lie_bracket end
+@doc "$(_doc_lie_bracket)"
+function lie_bracket(𝔤::LieAlgebra, X, Y)
+    Z = ManifoldsBase.allocate_result(𝔤, lie_bracket, X, Y)
+    return lie_bracket!(𝔤, Z, X, Y)
+end
+
+function lie_bracket! end
+@doc "$(_doc_lie_bracket)"
+lie_bracket!(𝔤::LieAlgebra, Z, X, Y)
+
+"""
+    is_point(𝔤::LieAlgebra, X; kwargs...)
+
+Check whether `X` is a valid point on the Lie Algebra `𝔤`.
+This falls back to checking whether `X` is a valid point on the tangent space
+at the [`identity_element``](@ref)`]`(G)` on `G.manifold` on the [`LieGroup`](@ref)
+of `G`
+"""
+function ManifoldsBase.is_point(𝔤::LieAlgebra, X; kwargs...)
+    # the manifold stored in the Fiber / Lie algebra is the Lie group G
+    G = 𝔤.manifold
+    e = identity_element(G)
+    return ManifoldsBase.is_vector(G.manifold, e, X; kwargs...)
+end
+
+# Move this line already to ManifoldsBase? On Fibers of course.
+LinearAlgebra.norm(𝔤::LieAlgebra, X) = LinearAlgebra.norm(𝔤.manifold, 𝔤.point, X)
+function LinearAlgebra.norm(
+    G::LieGroup{𝔽,O}, ::Identity{O}, X
+) where {𝔽,O<:AbstractGroupOperation}
+    return LinearAlgebra.norm(G, identity_element(G), X)
+end
```

Comment:
I'd expect `norm` for `VectorSpaceFiber` in `ManifoldsBase.jl` to already handle this.

Reply:
Or do you mean line 69? No that is not yet handled, but it would be nice to have that in ManifoldsBase.

Reply:
I'm not sure about the method in line 70 but at least the one in line 69 should be handled by https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/5972cde434c221009f354a229f7009856d4b0747/src/VectorFiber.jl#L11 .

Reply:
No. line 69 does not have an “outer `p`”. It just reads `norm(𝔤,X)` (not `norm(𝔤,p,X)`)

Reply:
Right, I didn't notice. Anyway, the method in line 69 could probably be introduced already in ManifoldsBase.jl for any `VectorSpaceFiber`?

Reply:
That is what line 68 is asking already, sure we could do that. _Then_ this line is really obsolete and we just need the l.70 for completeness.

Reply:
I've made a PR to ManifoldsBase.jl with that method.

Reply:
Nice. I noticed that we might not run Aqua here yet, I should add that!